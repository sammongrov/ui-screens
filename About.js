import React, { Component } from 'react';
import { Image, View, ScrollView, TouchableOpacity } from 'react-native';
import { styles } from 'react-native-theme';
import PropTypes from 'prop-types';
import { NavBar, Text, Screen, Icon, Button } from '@ui/components';
import { translate } from '../../../src/i18n';

export default class About extends Component {
  constructor(props) {
    super(props);
    const {
      appLogo,
      versionNo,
      buildNo,
      website,
      email,
      poweredBy,
      onHelpPress,
      onBackPress,
      navIconName,
      navIconColor,
      navIconSize,
      buttonTitle,
    } = this.props;

    this.appLogo = appLogo;
    this.versionNo = versionNo;
    this.buildNo = buildNo;
    this.website = website;
    this.email = email;
    this.poweredBy = poweredBy;
    this.onHelpPress = onHelpPress;
    this.onBackPress = onBackPress;
    this.navIconName = navIconName;
    this.navIconColor = navIconColor;
    this.navIconSize = navIconSize;
    this.buttonTitle = buttonTitle;
  }

  render() {
    const { appLogo } = this.props;
    return (
      <Screen>
        <NavBar
          leftComponent={
            <TouchableOpacity onPress={this.onBackPress}>
              <Icon name={this.navIconName} color={this.navIconColor} size={this.navIconSize} />
            </TouchableOpacity>
          }
          titleText="About"
        />
        <ScrollView contentContainerStyle={styles.scrollContentContainer}>
          <View style={[styles.centerColumnContainer, styles.padding15]}>
            <View style={styles.alignCenterJustifyEnd}>
              <Image source={appLogo} style={styles.logo} />
            </View>
            <View style={styles.centerContainer}>
              <Text style={[styles.subtext, styles.margin5]}>
                {translate('Version')}: {this.versionNo}
              </Text>
              <Text style={[styles.subtext3, styles.margin5]}>
                {translate('Build')}: {this.buildNo}
              </Text>
              <Button
                title={this.buttonTitle}
                onPress={this.onHelpPress}
                buttonStyle={[styles.btnPrimaryBg, styles.margin5]}
              />
              <Text style={[styles.subtext1, styles.margin5]}>{this.website}</Text>
              <Text style={[styles.subtext1, styles.margin5]}>{this.email}</Text>
              <Text style={[styles.subtext1, styles.margin5]}>{this.poweredBy}</Text>
            </View>
          </View>
        </ScrollView>
      </Screen>
    );
  }
}

About.propTypes = {
  versionNo: PropTypes.string.isRequired,
  buildNo: PropTypes.string.isRequired,
  website: PropTypes.string,
  email: PropTypes.string,
  poweredBy: PropTypes.string,
  appLogo: PropTypes.number.isRequired,
  onHelpPress: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  navIconName: PropTypes.string.isRequired,
  navIconColor: PropTypes.string.isRequired,
  navIconSize: PropTypes.number.isRequired,
  buttonTitle: PropTypes.string.isRequired,
};

About.defaultProps = {
  website: 'www.mongrov.com',
  email: 'support@mongrov.com',
  poweredBy: 'Powered by Mongrov, Inc.',
};
