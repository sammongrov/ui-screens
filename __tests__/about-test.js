import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from 'react-native';
import About from '../About';

const mockOnPress = jest
  .fn()
  .mockImplementationOnce(() => Alert.alert('Help is pressed'))
  .mockImplementationOnce(() => Alert.alert('Back is pressed'));

it('About renders correctly', () => {
  const tree = renderer
    .create(
      <About
        versionNo="12.35.67"
        buildNo="1.2.3"
        website="www.mongrov.com"
        email="support@mongrov.com"
        poweredBy="Powered by Mongrov, Inc."
        appLogo={777}
        onHelpPress={mockOnPress}
        onBackPress={mockOnPress}
        navIconName="icon1.jpg"
        navIconColor="#fff"
        navIconSize={30}
        buttonTitle="Press Me"
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
