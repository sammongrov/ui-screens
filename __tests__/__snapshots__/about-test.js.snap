// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ?
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            About
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView>
        <View>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View>
              <Image
                source={777}
              />
            </View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Version
                : 
                12.35.67
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Build
                : 
                1.2.3
              </Text>
              <View
                style={
                  Array [
                    Object {},
                    false,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#2089dc",
                          "borderRadius": 3,
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        Array [
                          undefined,
                          undefined,
                        ],
                        false,
                        false,
                        false,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "white",
                            "fontSize": 18,
                            "padding": 8,
                            "textAlign": "center",
                          },
                          undefined,
                          false,
                          false,
                        ]
                      }
                    >
                      Press Me
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                www.mongrov.com
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                support@mongrov.com
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Powered by Mongrov, Inc.
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
